client/extensions/woocommerce/app/settings/taxes/taxes-rates.js

```diff
@@ -0,0 +1,323 @@
+/**
+ * External dependencies
+ */
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { isEmpty, round } from 'lodash';
+import { localize } from 'i18n-calypso';
+import React, { Component, PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	areSettingsGeneralLoaded,
+	areTaxCalculationsEnabled,
+	getStoreLocation,
+} from 'woocommerce/state/sites/settings/general/selectors';
+import {
+	areTaxRatesLoaded,
+	getTaxRates,
+} from 'woocommerce/state/sites/meta/taxrates/selectors';
+import Card from 'components/card';
+import {
+	DESTINATION_BASED_SALES_TAX,
+	NO_SALES_TAX,
+	ORIGIN_BASED_SALES_TAX,
+} from 'woocommerce/lib/countries/constants';
+import { getCountryData, getStateData } from 'woocommerce/lib/countries';
+import ExtendedHeader from 'woocommerce/components/extended-header';
+import { fetchSettingsGeneral } from 'woocommerce/state/sites/settings/general/actions';
+import { fetchTaxRates } from 'woocommerce/state/sites/meta/taxrates/actions';
+import FormToggle from 'components/forms/form-toggle';
+import Notice from 'components/notice';
+import Table from 'woocommerce/components/table';
+import TableRow from 'woocommerce/components/table/table-row';
+import TableItem from 'woocommerce/components/table/table-item';
+
+class TaxesRates extends Component {
+
+	static propTypes = {
+		onEnabledChange: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			slug: PropTypes.string,
+		} ),
+	};
+
+	componentDidMount = () => {
+		const { address, loadedSettingsGeneral, loadedTaxRates, site } = this.props;
+
+		if ( site && site.ID ) {
+			if ( ! loadedSettingsGeneral ) {
+				this.props.fetchSettingsGeneral( site.ID );
+			} else if ( ! loadedTaxRates ) {
```

Comment:
Yep - we can't request the tax rates until we have the store address from settings general